### DISCUSSION TITLE
Is there any good way to clone Textbox?

### DISCUSSION THREAD

**Discussion by stack-guru** [fabricjs/fabric.js#8513] [Q&A]
textbox.clone(cloned => {
...
})
I am trying to clone textbox. So I used clone function. But this code is not running. 
Because text is null, In Textbox initialize function.
Is there any good way to clone Textbox? I am using fabric.4.6.0.js. Thanks.

**Comment by iocoker** [comment]
Clone should work fine, but you can also initialize a new one from the current object like this:

`const newText = new fabric.TextBox(textbox.text, { ...textbox.toObject() })`

**Reply by stack-guru** [reply]
Thanks. This is helpful.

**Comment by ShaMan123** [comment]
@stack-guru provide a repro so we can determine if this is a bug

**Reply by stack-guru** [reply]
Thanks for your care.

**Reply by ShaMan123** [reply]
It is not a matter of my care.
If you care about your ticket and getting answered, you should care and take a moment to investigate. Open source thrives only due to the effort of the community.
So in other words if you care about fabric, meaning you want it to function as expected you should own this.

**Reply by stack-guru** [reply]
ok.
Is clone function related to fromObject function?

**Reply by ShaMan123** [reply]
yes
you can see the source code of FabricObject#clone
